nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/context/TargetFactory.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.snmp.context;
+
+import org.apache.nifi.snmp.configuration.TargetConfiguration;
+import org.apache.nifi.snmp.utils.SNMPUtils;
+import org.snmp4j.AbstractTarget;
+import org.snmp4j.CommunityTarget;
+import org.snmp4j.UserTarget;
+import org.snmp4j.mp.SnmpConstants;
+import org.snmp4j.security.SecurityLevel;
+import org.snmp4j.smi.OctetString;
+import org.snmp4j.smi.UdpAddress;
+
+import java.util.Optional;
+
+public class TargetFactory {
```

Comment:
SNMPClientFactory uses:
- snmpVersion*
- securityName* (v3)
- authProtocol  (v3)
- authPassword (v3)
- privProtocol  (v3)
- privPassword (v3)

TargetFactory uses:
- snmpVersion*
- securityName* (v3)
- securityLevel (v3)
- communityString
- agentHost
- agentPort
- retries
- timeout

The * properties are common and the (v3) properties are only used in case of SNMPv3.

Reply:
I can create different Config classes, 
- SnmpV1V2cConfig extends SnmpConfig
- SnmpV3Config extends SnmpConfig

same with TargetConfig.